.github/workflows/_docker-build.yml

```diff
@@ -17,51 +24,59 @@ jobs:
     steps:
       - name: Set UUID for the runner
         id: set-uuid
-        run: echo "::set-output name=runner-uuid::$(uuidgen)"
+        run: echo "runner-uuid=$(uuidgen)" >> $GITHUB_OUTPUT
 
-  build-and-run:
-    name: Build, test, and run the runner image
+  build-and-test:
+    name: Build and test the runner image
     runs-on: ubuntu-latest
     needs: set-runner-uuid
     outputs:
-      sha-tag: ${{ steps.set-sha-tag.outputs.sha-tag }}
+      image-tag-with-sha: ${{ steps.set-image-tag-with-sha.outputs.image-tag-with-sha }}
+      image-tag-with-repo: ${{ steps.set-image-tag-with-repo.outputs.image-tag-with-repo }}
     steps:
       - name: Check out repo
         uses: actions/checkout@v3
 
       - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v3
+        uses: aws-actions/configure-aws-credentials@v4
         with:
           aws-region: us-east-1
           role-to-assume: ${{ secrets.OIDC_IAM_ROLE_ARN }}
 
       - name: Log in to ECR
         id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+        uses: aws-actions/amazon-ecr-login@v2
 
-      - name: Set short SHA for tag
-        id: set-sha-tag
-        run: echo "::set-output name=sha-tag::$(git rev-parse --short "$GITHUB_SHA")"
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-      - name: Set SHA tag for Docker image
-        run: echo "SHA_TAG=${{ steps.login-ecr.outputs.registry }}/github-actions-runner:${{ steps.set-sha-tag.outputs.sha-tag }}" >> $GITHUB_ENV
+      - name: Set image tag with short SHA
+        id: set-image-tag-with-sha
+        run: echo "image-tag-with-sha=${{ inputs.image_tag }}-$(git rev-parse --short "$GITHUB_SHA")" >> $GITHUB_OUTPUT
 
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+      - name: Set image tag with ECR repo
+        id: set-image-tag-with-repo
+        run: echo "image-tag-with-repo=${{ steps.login-ecr.outputs.registry }}/github-actions-runner:${{ steps.set-image-tag-with-sha.outputs.image-tag-with-sha }}" >> $GITHUB_OUTPUT
+
+      - name: Set Docker env variables
+        run: cat docker.env >> $GITHUB_ENV
 
       - name: Build Docker image
         uses: docker/build-push-action@v4
         with:
           context: .
-          load: true # export a tarball so we can test the image locally before pushing
-          tags: ${{ env.SHA_TAG }}
-          cache-from: type=gha # use the GitHub Cache API
+          file: ${{ inputs.dockerfile }}
+          tags: ${{ steps.set-image-tag-with-repo.outputs.image-tag-with-repo }}
+          build-args: ACTIONS_VERSION
+          cache-from: type=gha
           cache-to: type=gha,mode=max
+          load: true # export a tarball so we can run the image tests locally before pushing to ECR
```

Comment:
Right, which means "Will automatically load the single-platform build result to docker images." Not that it exports a tarball. Or I am misunderstanding maybe?

Reply:
I think tarball is `type=tar` https://docs.docker.com/engine/reference/commandline/buildx_build/#output

Reply:
Okay, I find that documentation very confusing and maybe there is in fact a tarball involved in `type=docker` as well, although it may not be written out to the normal filesystem? Anyway my impression from the rest of the workflow is that the important part is that it gets loaded to the local docker instance; the later commands (scan, lint, and push) expect the image to be located by tag in the local docker instance, not for there to be a path to a tarball. Which is why I was confused by a comment suggesting that we need a tarball for the future steps. Does that make sense?